### PR TITLE
Migrate .NET 3.1 and PHP 7.4 runtime images to bullseye flavor

### PR DESCRIPTION
**Pull request by cormacpayne**
Resolves work item 1541627 -- migrates the existing .NET 3.1 and PHP 7.4 runtime images to the bullseye flavor of Debian.

- [x] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as \#\<issue\_number\>.
~- [ ] Tests are included and/or updated for code changes.~
~- [ ] Proper license headers are included in each file.~

### CODE REVIEW

File: vsts/scripts/pushBaseImagesToRegistry.sh [hunk 1]
```diff
@@ -32,3 +32,11 @@ if [ -f "$stretchBaseImagesFile" ]; then
...
 fi
 
+if [ -f "$bullseyeBaseImagesFile" ]; then
+    echo "$bullseyeBaseImagesFile exists. pushing bullseye image tags ..."
+    while read imageName; do
+      # read the tags from file
+      echo "Pushing image $imageName ..."
+      docker push "$imageName"
+    done <"$bullseyeBaseImagesFile"
+fi
```
Comment (william-msft) [inline vsts/scripts/pushBaseImagesToRegistry.sh:42]: Do the tests that we currently have test that the platforms build and run successfully with the new Bullseye image?

Comment (cormacpayne) [reply vsts/scripts/pushBaseImagesToRegistry.sh:42]: There are existing runtime tests that expect specific versions of the platform runtimes to be built and run tests against them -- this includes the two versions that have been migrated in this PR. However, I'm not sure if there's a way to test what Debian flavor we're using within the runtime image, so maybe that's something we can investigate.

File: images/runtime/php/__versions.sh [hunk 2]
```diff
@@ -3,4 +3,5 @@
 declare -r __DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
 source "$__DIR/../../../build/__phpVersions.sh"
 declare -r VERSION_ARRAY=($PHP73_VERSION $PHP72_VERSION $PHP70_VERSION $PHP56_VERSION)
-declare -r VERSION_ARRAY_BUSTER=($PHP74_VERSION $PHP80_VERSION $PHP81_VERSION)
\ No newline at end of file
+declare -r VERSION_ARRAY_BUSTER=($PHP80_VERSION $PHP81_VERSION)
+declare -r VERSION_ARRAY_BULLSEYE=($PHP74_VERSION)
```
Comment (qianz2) [inline images/runtime/php/__versions.sh:7]: Can we have tests for PHP 7.4 bullseye based runtime image? I'm not sure if we need to modify anything inside that image other than renaming the dockerfile. It's good to add tests to make sure everything works fine.

**Review by qianz2 — DISMISSED** [review]

**Review by qianz2 — APPROVED** [review]

File: images/runtime/php/7.4/base.bullseye.Dockerfile [hunk 3]
```diff
@@ -19,6 +19,12 @@ RUN echo -e '<FilesMatch "\.(?i:ph([[p]?[0-9]*|tm[l]?))$">\n SetHandler applicat
 # Disable Apache2 server signature
 RUN echo -e 'ServerSignature Off' >> /etc/apache2/apache2.conf
 RUN echo -e 'ServerTokens Prod' >> /etc/apache2/apache2.conf
+RUN { \
+   echo '<DirectoryMatch "^/.*/\.git/">'; \
+   echo '   Order deny,allow'; \
+   echo '   Deny from all'; \
+   echo '</DirectoryMatch>'; \
+} >> /etc/apache2/apache2.conf
```
Comment (qianz2) [inline images/runtime/php/7.4/base.bullseye.Dockerfile:27]: Does this command required when migrating to bullseye, or it is a missing part generally?

Comment (cormacpayne) [reply images/runtime/php/7.4/base.bullseye.Dockerfile:27]: As discussed offline, this change resulted from running the `buildRuntimeBaseImages.sh` script.

**Pull request merged**